### PR TITLE
fix: strip ansi-colors to get the branch name correctly

### PR DESCRIPTION
**Pull request by jtakalai**
I got an error like this:
```
$ git commit
.husky/commit-msg: line 4: [: too many arguments
```
the commit went through but without the check

the reason I think is that my `git branch` adds colors, so this new sed magic now strips that ansi-coloring.

### CODE REVIEW

File: .husky/commit-msg [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
-BRANCH_NAME=$(git branch | grep '*' | sed 's/* //')
+BRANCH_NAME=$(git branch | grep '*' | sed 's/.*\*.* //')
```
Comment (kare) [inline .husky/commit-msg:2]: I think it should be: `git branch --no-color`

Comment (timoxley) [reply .husky/commit-msg:2]: Nice one @kare

**Review by timoxley — CHANGES_REQUESTED** [review]
if Kare's `--no-color` works then we should probably do that

**Comment by jtakalai** [comment]
it was `grep` putting the color in there (highlighting the "`*`" it found)

File: .husky/commit-msg [hunk 2]
```diff
@@ -1,5 +1,5 @@
 #!/bin/sh
-BRANCH_NAME=$(git branch | grep '*' | sed 's/* //')
+BRANCH_NAME=$(git branch --no-color | grep --color=never '*' | sed 's/* //')
```
Comment (kare) [inline .husky/commit-msg:2]: Looks like my `git branch` solution is not the correct one. While scripting with Git plumbing commands should be used and `git branch` is not a plumbing but porcelain command.
https://git-scm.com/book/en/v2/Git-Internals-Plumbing-and-Porcelain
https://git-blame.blogspot.com/2013/06/checking-current-branch-programatically.html

TLDR: `git symbolic-ref --short -q HEAD`

Comment (jtakalai) [reply .husky/commit-msg:2]: ok! maybe that one then

**Review by kare — APPROVED** [review]
LGTM
